frontend/public/components/modals/rollback-modal.jsx

```diff
@@ -0,0 +1,215 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  getDeploymentConfigNameForRC,
+  getDeploymentConfigVersion,
+  getDeploymentNameForRS,
+} from '@console/shared/src';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { LoadingInline, withHandlePromise } from '../utils';
+import { DeploymentConfigModel, DeploymentModel, ReplicationControllerModel } from '../../models';
+import { k8sCreate, k8sPatch, k8sUpdate } from '../../module/k8s';
+import { useK8sWatchResource } from '../utils/k8s-watch-hook';
+import { Alert } from '@patternfly/react-core';
+
+const ANNOTATIONS_TO_SKIP = [
+  'kubectl.kubernetes.io/last-applied-configuration',
+  'deployment.kubernetes.io/revision',
+  'deployment.kubernetes.io/revision-history',
+  'deployment.kubernetes.io/desired-replicas',
+  'deployment.kubernetes.io/max-replicas',
+  'deprecated.deployment.rollback.to',
+];
+
+const BaseRollbackModal = withHandlePromise((props) => {
+  const { t } = useTranslation();
+  const isDCRollback = props.resource.kind === ReplicationControllerModel.kind;
+  const dName = isDCRollback
+    ? getDeploymentConfigNameForRC(props.resource)
+    : getDeploymentNameForRS(props.resource);
+  const [changeScaleSettings, setChangeScaleSettings] = React.useState(false);
+  const [changeStrategy, setChangeStrategy] = React.useState(false);
+  const [changeTriggers, setChangeTriggers] = React.useState(false);
+  const deploymentResource = {
+    kind: isDCRollback ? DeploymentConfigModel.kind : DeploymentModel.kind,
```

Comment:
Do we need an API group here? I'm a little surprised this works.

Reply:
It really doesn't like when I use `referenceForModel(DeploymentModel)` or `referenceForModel(DeploymentConfigModel)`

Reply:
I think there's probably an existing bug here somewhere :/ I'd expect that to work, and we should really be using the API group for anything that has one. But we can look at that separately from this PR.